### PR TITLE
chore: changed the datasource-update event tag for analytics

### LINKED ISSUE #17099: [Task]: Differentiate between server side analytics for save datasource flow and other updates on datasource
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Today when a datsource is saved by a user by clicking on the Save button in the datasource editor, the server sends an update_DATASOURCE event to analytics. However, this same event is triggered even when the datasource gets updated by the backend as a result of refreshing tokens. We need to separate this event, by either pathing the code flow differently, or introducing a new event to only track these invisible updates so that we can remove them from our metrics.

### PR DESCRIPTION
**Pull request by sondermanish**
## Description

Previously we are using the `isDatasourceUpdateServerInvoked` tag for distinguishing the datasource update event b/w user and server.  The tag was less intuitive hence flipped the tag, now the tag is  `isDatasourceUpdateUserInvoked` which is `True` when a user call triggered this event, `False`  otherwise.   

Fixes #17099

## Type of change

- Chore

## How Has This Been Tested?
- Manually

This doesn't require additional tests as the existing tests are adequate. Please add `testPlanApproved` tag

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java [hunk 1]
```diff
@@ -229,13 +229,13 @@ public Mono<Datasource> update(String id, Datasource datasource, Boolean isServe
                 .flatMap(savedDatasource -> {
 
                     // this key will present in the analytics as a diff b/w server and user invoked flows
-                    String isDatasourceUpdateServerInvokedKey = "isDatasourceUpdateServerInvoked";
+                    String isDatasourceUpdateUserInvokedKey = "isDatasourceUpdateUserInvoked";
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/DatasourceServiceCEImpl.java:232]: Nit: Can we pull this away into a constant?

**Comment by sondermanish** [comment]
ok-to-test sha=9ecd461

**Review by nidhi-nair — APPROVED** [review]

**Comment by sondermanish** [comment]
Auto Approving the test plan, as the test plans are adequate.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**